### PR TITLE
Tweak build.sbt to support MacOS ARM builds

### LINKED ISSUE #2: Known issue with Mac M1?
Hi !

Super cool to have Scala go to bare metal!

I tried it on my amd64 machine (MacOS) works great there, but also tried it on my new M1 and during `sbt run` it fails with

```
[info] welcome to sbt 1.7.1 (Homebrew Java 18.0.1.1)
[info] loading global plugins from /Users/markie/.sbt/1.0/plugins
[info] loading settings for project scala-native-ember-example-build from plugins.sbt ...
[info] loading project definition from /Users/markie/Projects/scala-native-ember-example/project
[info] loading settings for project scala-native-ember-example from build.sbt ...
[info] set current project to Scala Native Ember Example (in build file:/Users/markie/Projects/scala-native-ember-example/)
[info] Linking (2639 ms)
[info] Discovered 6743 classes and 44040 methods
[info] Optimizing (debug mode) (4985 ms)
[info] Generating intermediate code (7295 ms)
[info] Produced 10 files
[info] Compiling to native code (3662 ms)
[error] ld: warning: directory not found for option '-L/usr/local/opt/openssl@1.1/lib'
[error] ld: library not found for -lcrypto
[error] clang: error: linker command failed with exit code 1 (use -v to see invocation)
[error] Failed to link /Users/markie/Projects/scala-native-ember-example/target/scala-3.2.0/scala-native-ember-example-out
[error] (Compile / nativeLink) Failed to link /Users/markie/Projects/scala-native-ember-example/target/scala-3.2.0/scala-native-ember-example-out
```

I installed `brew install gsl openssl@1.1 s2n` .. but turns out `openssl version` returns `LibreSSL 2.8.3` which might be causing trouble ?

Is this something known?

### PR DESCRIPTION
**Pull request by Fristi**
Fixes #2

### CODE REVIEW

**Comment by Fristi** [comment]
Thanks to @armanbilge!

File: build.sbt [hunk 1]
```diff
@@ -15,20 +15,16 @@ libraryDependencies ++= Seq(
...
+val isMacOs = Option(System.getProperty("os.name")).exists(_.toLowerCase().contains("mac"))
+val isArm = Option(System.getProperty("os.arch")).exists(_.toLowerCase().contains("aarch64"))
 
 nativeConfig ~= { c =>
   if (isLinux) { // brew-installed s2n
     c.withLinkingOptions(c.linkingOptions :+ "-L/home/linuxbrew/.linuxbrew/lib")
   } else if (isMacOs) // brew-installed OpenSSL
-    c.withLinkingOptions(c.linkingOptions :+ "-L/usr/local/opt/openssl@1.1/lib")
+    if(isArm) c.withLinkingOptions(c.linkingOptions :+ "-L/opt/homebrew/opt/openssl@3/lib")
+    else c.withLinkingOptions(c.linkingOptions :+ "-L/usr/local/opt/openssl@1.1/lib")
```
Comment (armanbilge) [inline build.sbt]: We should update this one as well, due to the s2n update.
```suggestion
    else c.withLinkingOptions(c.linkingOptions :+ "-L/usr/local/opt/openssl@3/lib")
```

**Review by armanbilge — COMMENTED** [review]
Thank you so much for trying this and opening the PR!

**Review by armanbilge — APPROVED** [review]
Thanks!!

**Pull request merged**